shared/cert.go

```diff
@@ -32,6 +32,7 @@ import (
 // CertOptions holds configuration for creating a new CertInfo.
 type CertOptions struct {
 	AddHosts    bool
+	SubjectName string
```

Comment:
Should we add a comment saying if `SubjectName` isn't specified the hostname is used instead?

Reply:
Please can these fields have documentation comments added to them explaining what they do and when.